langchain/vectorstores/qdrant.py

```diff
@@ -226,10 +227,21 @@ def similarity_search_with_score(
             List of Documents most similar to the query and score for each.
         """
 
+        if filter is not None and isinstance(filter, dict):
+            warnings.warn(
```

Comment:
are we sure we want to warn here? if dicts are sufficient for some folks is there anything wrong with them keeping as is?

Reply:
@dev2049 The dict-based filtering covers some basic cases only and is not well-defined, to be honest. Recently there have been some changes introduced that made it even more complicated and I believe we should eventually get rid of them. I think there should be a single way exposed unless you have the plan to unify it for all the vector stores.